src/main/java/io/fabric8/jenkins/openshiftsync/BuildConfigWatcher.java

```diff
@@ -96,6 +97,18 @@
   private Watch buildConfigWatch;
   private ScheduledFuture relister;
 
+  private static final ConcurrentHashSet<String> deletesInProgress = new ConcurrentHashSet<String>();
+
+  public static void deleteInProgress(String bcName) {
```

Comment:
was there a reason you did not make a call to this method in your `deleteJob` method below? i.e. is your intent only to put some of the scaffolding for this fix in, or did you just overlook adding the call to this?

If the former, put some comments here indicating more changes are needed to fully activate.